quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_parsing.rs

```diff
@@ -93,27 +99,25 @@ fn parse_strftime_with_tz(
 }
 
 /// Returns the appropriate tantivy datetime for the specified unix timestamp.
-pub(super) fn parse_timestamp(
-    timestamp: i64,
-    date_time_formats: &[DateTimeFormat],
-) -> Result<TantivyDateTime, String> {
-    for date_time_format in date_time_formats {
-        let date_time = match date_time_format {
-            DateTimeFormat::TimestampSecs => TantivyDateTime::from_timestamp_secs(timestamp),
-            DateTimeFormat::TimestampMillis => TantivyDateTime::from_timestamp_millis(timestamp),
-            DateTimeFormat::TimestampMicros => TantivyDateTime::from_timestamp_micros(timestamp),
-            DateTimeFormat::TimestampNanos => {
-                TantivyDateTime::from_timestamp_micros(timestamp / 1_000)
-            }
-            _ => continue,
-        };
-        return Ok(date_time);
+///
+/// This function will choose the timestamp precision based on the
+/// most significant bit position in the timestamp value.
+/// The tradeoff is that we can only support dates ranging from
+/// `01 Jan 1974 00:00:00` to `31 Dec 2241 23:59:59`
+pub(super) fn parse_timestamp(timestamp: i64) -> Result<TantivyDateTime, String> {
+    if !(MIN_TIMESTAMP_SECONDS..=MAX_TIMESTAMP_NANOSECONDS).contains(&timestamp) {
+        return Err(format!(
+            "Failed to parse unix timestamp `{timestamp}`. Quickwit only support timestamp values \
+             ranging form `01 Jan 1974 00:00:00` to `31 Dec 2241 23:59:59`."
+        ));
+    }
+    match i64::BITS - timestamp.leading_zeros() {
+        27..=33 => Ok(TantivyDateTime::from_timestamp_secs(timestamp)),
+        37..=43 => Ok(TantivyDateTime::from_timestamp_millis(timestamp)),
+        47..=53 => Ok(TantivyDateTime::from_timestamp_micros(timestamp)),
+        57..=63 => Ok(TantivyDateTime::from_timestamp_micros(timestamp / 1_000)),
+        _ => unreachable!(),
```

Comment:
We must protect ourselves against bogus data not covered by those ranges: for instance, `MIN_TIMESTAMP_SECONDS << 8` will panic.

Reply:
Humm nice catch :+1: , I actually checked against these edge cases and this is why I put `unreachable` but I used a wrong data (timestamps).